WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayInformationViewController.swift

```diff
@@ -0,0 +1,189 @@
+import SwiftUI
+
+/// This view controller is used when no reader is connected. It assists
+/// the merchant in connecting to a reader.
+///
+final class SetUpTapToPayInformationViewController: UIHostingController<SetUpTapToPayInformationView>, PaymentSettingsFlowViewModelPresenter {
+
+    private var viewModel: SetUpTapToPayInformationViewModel
+
+    private lazy var alertsPresenter = CardPresentPaymentAlertsPresenter(rootViewController: self)
+
+    /// Connection Controller (helps connect readers)
+    ///
+    private lazy var connectionController: BuiltInCardReaderConnectionController = {
+        return BuiltInCardReaderConnectionController(
+            forSiteID: viewModel.siteID,
+            alertsPresenter: alertsPresenter,
+            alertsProvider: BuiltInReaderConnectionAlertsProvider(),
+            configuration: viewModel.configuration,
+            analyticsTracker: viewModel.connectionAnalyticsTracker)
+    }()
```

Comment:
I get where you're coming from with switching sites, but this VC can't be retained while switching sites, so it won't happen in practice.

The existing setup is not ideal, and it needs refactored on its source too (the Manage card readers VC).

Computed won't work though, because we need to keep hold of the connection controller so it can... do the whole connection! It's lazy because of a chicken and egg problem in the initialiser (needing other stuff from self.)

The fix I'd like to move towards is have the ConnectionController be a dependency of the ViewModel, and all the actions go through there. Too big for this PR but I'll look to do it in future.

Reply:
Thanks for the explanation, that makes sense!